src/ForgotPasswordController.js

```diff
@@ -93,72 +105,73 @@ function (Okta, FormController, Enums, FormType, ValidationUtil, ContactSupport,
-                else {
-                  mobileFactors = Okta.loc('recovery.sms');
-                }
-                return { mobileFactors : mobileFactors };
-              }
-            })
           }));
+          if (smsEnabled || callEnabled) {
+            formChildren.push(FormType.View({
+              View: Okta.View.extend({
+                template: '\
+                  <p class="mobile-recovery-hint">\
+                    {{i18n code="recovery.mobile.hint" bundle="login" arguments="mobileFactors"}}\
+                  </p>',
+                getTemplateData: function () {
+                  var mobileFactors;
+                  if (smsEnabled && callEnabled) {
+                    mobileFactors = Okta.loc('recovery.smsOrCall');
+                  }
+                  else if (callEnabled) {
+                    mobileFactors = Okta.loc('recovery.call');
+                  }
+                  else {
+                    mobileFactors = Okta.loc('recovery.sms');
+                  }
+                  return { mobileFactors : mobileFactors };
+                }
+              })
+            }));
+          }
+          if(smsEnabled) {
+            this.$el.addClass('forgot-password-sms-enabled');
+            formChildren.push(this.addRecoveryFactorButton('sms-button', 'password.forgot.sendText',
+              Enums.RECOVERY_FACTOR_TYPE_SMS, this));
+          }
+          if(callEnabled) {
+            this.$el.addClass('forgot-password-call-enabled');
+            formChildren.push(this.addRecoveryFactorButton('call-button', 'password.forgot.call',
+              Enums.RECOVERY_FACTOR_TYPE_CALL, this));
+          }
+          if(emailEnabled) {
+            this.$el.addClass('forgot-password-email-enabled');
+            formChildren.push(this.addRecoveryFactorButton('email-button', 'password.forgot.sendEmail',
+              Enums.RECOVERY_FACTOR_TYPE_EMAIL, this));
+          }
         }
 
         return formChildren;
       },
       initialize: function () {
-        var form = this;
-
-        if (this.settings.get('features.callRecovery')) {
-          this.$el.addClass('forgot-password-call-enabled');
-          this.addRecoveryFactorButton('call-button', 'password.forgot.call',
-            Enums.RECOVERY_FACTOR_TYPE_CALL, form);
-        }
-        if (this.settings.get('features.smsRecovery')) {
-          this.$el.addClass('forgot-password-sms-enabled');
-          this.addRecoveryFactorButton('sms-button', 'password.forgot.sendText',
-            Enums.RECOVERY_FACTOR_TYPE_SMS, form);
-        }
 
         this.listenTo(this.state, 'contactSupport', function () {
           this.add(ContactSupport, '.o-form-error-container');
         });
 
         this.listenTo(this, 'save', function () {
           this.options.appState.set('username', this.model.get('username'));
+          this.model.save();
         });
       },
       addRecoveryFactorButton: function (className, labelCode, factorType, form) {
```

Comment:
nit: rename this to ```createRecoveryFactorButton``` or something similar, since this function no longer adds a button